internal/controllers/machineset/machineset_controller.go

```diff
@@ -389,6 +364,43 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 	return ctrl.Result{}, nil
 }
 
+// syncMachines updates Machines to propagate in-place mutable fields from the MachineSet.
+// Note: It also cleans up managed fields of all Machines so that Machines that were created/patched before the
+// controller adopted SSA can also work with SSA.  Otherwise fields would be co-owned by our "old" "manager" and
+// "capi-machineset" and then we would not be able to e.g. drop labels and annotations.
+// TODO: update the labels and annotations to the corresponding infra machines and the boostrap configs of the filtered machines.
```

Comment:
is this a TODO for this PR or for later? if it's for later, is it better to track in an issue?

Reply:
It's part of https://github.com/kubernetes-sigs/cluster-api/issues/7731 the WIP PR is here: https://github.com/kubernetes-sigs/cluster-api/pull/8060

TODO is mostly there additionally to signal that it will be added in this func